components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -31,133 +31,191 @@ type Props = PropsFromRedux & OwnProps;
                 onClick={handleUnfavoriteChannel}
-                icon={<StarIcon size={16}/>}
-                text={formatMessage({id: 'sidebar_left.sidebar_channel_menu.unfavoriteChannel', defaultMessage: 'Unfavorite'})}
+                leadingElement={<StarIcon size={18}/>}
+                labels={
+                    <FormattedMessage
+                        id='sidebar_left.sidebar_channel_menu.unfavoriteChannel'
+                        defaultMessage='Unfavorite'
+                    />
+                }
             />
         );
     } else {
-        function handleFavoriteChannel() {
+        function handleFavoriteChannel(event: MouseEvent<HTMLLIElement>) {
+            event.preventDefault();
+
             props.favoriteChannel(props.channel.id);
             trackEvent('ui', 'ui_sidebar_channel_menu_favorite');
         }
 
-        favoriteMenuItem = (
-            <Menu.ItemAction
+        favoriteUnfavoriteMenuItem = (
+
+            <Menu.Item
                 id={`favorite-${props.channel.id}`}
                 onClick={handleFavoriteChannel}
-                icon={<StarOutlineIcon size={16}/>}
-                text={formatMessage({id: 'sidebar_left.sidebar_channel_menu.favoriteChannel', defaultMessage: 'Favorite'})}
+                leadingElement={<StarOutlineIcon size={18}/>}
+                labels={
+                    <FormattedMessage
+                        id='sidebar_left.sidebar_channel_menu.favoriteChannel'
+                        defaultMessage='Favorite'
+                    />
+                }
             />
         );
     }
 
-    let muteChannelMenuItem: JSX.Element | null = null;
+    let muteUnmuteChannelMenuItem: JSX.Element | null = null;
     if (props.isMuted) {
-        let muteChannelText = formatMessage({id: 'sidebar_left.sidebar_channel_menu.unmuteChannel', defaultMessage: 'Unmute Channel'});
+        let muteChannelText = (
+            <FormattedMessage
+                id='sidebar_left.sidebar_channel_menu.unmuteChannel'
+                defaultMessage='Unmute Channel'
+            />
+        );
         if (props.channel.type === Constants.DM_CHANNEL || props.channel.type === Constants.GM_CHANNEL) {
-            muteChannelText = formatMessage({id: 'sidebar_left.sidebar_channel_menu.unmuteConversation', defaultMessage: 'Unmute Conversation'});
+            muteChannelText = (
+                <FormattedMessage
+                    id='sidebar_left.sidebar_channel_menu.unmuteConversation'
+                    defaultMessage='Unmute Conversation'
+                />
+            );
         }
 
-        function handleUnmuteChannel() {
+        function handleUnmuteChannel(event: MouseEvent<HTMLLIElement>) {
+            event.preventDefault();
+
             props.unmuteChannel(props.currentUserId, props.channel.id);
         }
 
-        muteChannelMenuItem = (
-            <Menu.ItemAction
+        muteUnmuteChannelMenuItem = (
```

Comment:
same structure as above would also clean this up nicely

Reply:
Hey, Thank you for the feedback, but i would like to leave this as is. Hope you don't mind.

Reply:
same as above